docs/rules/no-type-literal.md

```diff
@@ -0,0 +1,121 @@
+# Enforces the use of interfaces over type literals (`type T = {...}`).
```

Comment:
by the way, the allowUnion and allowIntersection enable compositions that include a literal type. e.g.
`type foo = string & {};`

now I'm not so sure if we still need them if we go with literals and callbacks. I think we can rename the rule to no-type-aliases, with two properties allowTypeLiterals and allowCallbacks